Controllers/SightingControllers.js

```diff
@@ -0,0 +1,30 @@
+class SightingsController {
+  constructor(getSightings) {
+    this.getSightings = getSightings;
+  }
+
+  getAllData = async (req, res) => {
+    const sightings = await this.getSightings();
+    res.json(sightings);
+  };
+
+  getOneData = async (req, res) => {
+    const { sightingIndex } = req.params;
+    if (isNaN(Number(sightingIndex))) {
+      return res.status(400).send({ error: true, msg: "Wrong Type" });
+    }
+    const sightings = await this.getSightings();
+    res.json(sightings[sightingIndex]);
+  };
+
+  getFilterData = async (req, res) => {
```

Comment:
```suggestion
  getByFilter = async (req, res) => {
```

Reply:
Very nice function with dynamic filter btw. Usually I would however allow multiple filter parameters, and then apply those one by one.
Example:
`endpoint/url?year=1992&month=january&day=1`

and then I would apply all three filters here to allow as fine-grained filtering as the user wants. With your implementation only one such filter is possible at a given time